volmdlr/faces.py

```diff
@@ -2641,6 +2651,20 @@ def neutral_fiber(self):
         point2 = self.surface3d.frame.origin + self.surface3d.frame.w * zmax
         return volmdlr.wires.Wire3D([vme.LineSegment3D(point1, point2)])
 
+    def get_normal_at_point(self, point):
+        """
+        Gets Normal vector at a given point on the face.
+        :param point: point on the face.
+        :return:
+        """
+        if not self.point_belongs(point):
+            raise ValueError(f'Point {point} not in this face.')
+        vector = (- point).to_point()
+        p_vector = (point - self.surface3d.frame.origin).to_vector()
+        t_param = p_vector.dot(self.surface3d.frame.w) / self.surface3d.frame.w.dot(self.surface3d.frame.w)
+        proj_point = self.surface3d.frame.origin + t_param * self.surface3d.frame.w
+        return (proj_point - point).unit_vector()
```

Comment:
you can ask the surface its normal vector. For the cylindrical surface, for exemple the normal vector at a given point on the surface is given by:

> theta, _ =  self.point3d_to_2d(point)
> normal = math.cos(theta) * self.frame.u + math.sin(theta) * self.frame.v

see, for more detail and other surfaces normal formulas: https://www.steptools.com/stds/smrl/data/resource_docs/geometric_and_topological_representation/sys/4_schema.htm#geometry_schema.cylindrical_surface